STX.REST.RESTFulSense.Clients.Tests.Unit/Services/Foundations/HttpExchanges/HttpExchangeServiceTests.Validations.Get.cs

```diff
@@ -0,0 +1,298 @@
+using FluentAssertions;
+using Moq;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+using Xunit;
+
+namespace STX.REST.RESTFulSense.Clients.Tests.Unit.Services.Foundations.HttpExchanges
+{
+    public partial class HttpExchangeServiceTests
+    {
+        [Fact]
+        private async Task ShouldThrowHttpExchangeValidationExceptionOnGetIfHttpExchangeIsNullAsync()
+        {
+            // given
+            HttpExchange nullHttpExchange = null;
+
+            var nullHttpExchangeException =
+                new NullHttpExchangeException(
+                    message: "Null HttpExchange error occurred, fix errors and try again.");
+
+            var expectedHttpExchangeValidationException =
+                new HttpExchangeValidationException(
+                    message: "HttpExchange validation errors occurred, fix errors and try again.",
+                    innerException: nullHttpExchangeException);
+
+            // when
+            ValueTask<HttpExchange> getAsyncTask =
+                httpExchangeService.GetAsync(nullHttpExchange);
+
+            HttpExchangeValidationException actualHttpExchangeValidationException =
+                await Assert.ThrowsAsync<HttpExchangeValidationException>(
+                    getAsyncTask.AsTask);
+
+            //then
+            actualHttpExchangeValidationException.Should().BeEquivalentTo(
+                expectedHttpExchangeValidationException);
+
+            this.httpBroker.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        private async Task ShouldThrowHttpExchangeValidationExceptionOnGetIfHttpExchangeRequestIsNullAsync()
+        {
+            // given
+            var httpExchange = new HttpExchange();
+
+            var invalidHttpExchangeRequestException = new InvalidHttpExchangeRequestException(
+                message: "Invalid HttpExchange request error occurred, fix errors and try again.");
+
+            invalidHttpExchangeRequestException.UpsertDataList(
+                key: nameof(HttpExchange.Request),
+                value: "Value is required");
+
+            var expectedHttpExchangeValidationException = new HttpExchangeValidationException(
+                message: "HttpExchange validation errors occurred, fix errors and try again.",
+                innerException: invalidHttpExchangeRequestException);
+
+            // when
+            ValueTask<HttpExchange> getAsyncTask = httpExchangeService.GetAsync(httpExchange);
+
+            HttpExchangeValidationException actualHttpExchangeValidationException =
+                await Assert.ThrowsAsync<HttpExchangeValidationException>(
+                    getAsyncTask.AsTask);
+
+            //then
+            actualHttpExchangeValidationException.Should().BeEquivalentTo(
+                expectedHttpExchangeValidationException);
+
+            this.httpBroker.VerifyNoOtherCalls();
+        }
+
+        [Theory]
```

Comment:
why removing some input parameters for covering if consumer set a POST method when calling the GetAsync?